### PR TITLE
Obsolete retool kits in mods

### PR DESCRIPTION
**Pull request by smolbird**
Finishes incomplete implementation of https://github.com/CleverRaven/Cataclysm-DDA/pull/23078

Moves retool items from Icecoon's Weapons Pack and Expanded Realistic Guns to obsolete.json, removing recipes to craft them.

Side note, do we even have any mods left that use the bore slot without these? I'd assume more sensible, weapon-specific conversion kits are planned for the future to use that slot?

### CODE REVIEW

**Comment by ymber** [comment]
I left the bore slot available to allow for weapon specific mods in the future. It should be trivial to implement support for them. That said, I would like to do proper weapon modularity which would abolish weapon mods in favor of a parts system, but that's a much bigger project.

**Comment by smolbird** [comment]
That's what I guessed, yeah.

**Comment by cainiaowu** [comment]
You also need to add something in migration.json so it doesnt break saves.

**Comment by smolbird** [comment]
>You also need to add something in migration.json so it doesnt break saves.

I wasn't aware of this, and @ymber neglected that too in their PR.

**Comment by smolbird** [comment]
What do I migrate the items into though? I'm fairly certain obsolete.json is in fact how you retain items that shouldn't show up anymore, by having them still exist in an unused file.

**Comment by ymber** [comment]
Migrations are for changing object id strings. Shifting definitions unmodified into another file doesn't require any migrations.

**Comment by kevingranade** [comment]
The previous PR didn't remove the item definitions, it just kept them from spawning.

**Comment by smolbird** [comment]
>Should these be moved into mainline obsolete.json or an obsolete.json in the mod folder?

A 6.54mm item from Icecoon's Weapons Pack was already present in obsolete.json, so practice seems to be move to mainline obsolete file.

> The previous PR didn't remove the item definitions, it just kept them from spawning.

Same here, mod retool kits were moved to obsolete.json.

**Comment by kevingranade** [comment]
TL;DR: this is fine.

obsolete.json is just a name, it doesn't actually change how the game loads the items, it just makes it clear that the items are in the process of going away.  As such , no saves should be damaged by this change.

I'm not too concerned about which obsolete.json they end up in.

**Comment by smolbird** [comment]
Sorry it took a while, internet issues when I got home.

**Comment by Night-Pryanik** [comment]
By the way, why do we do this in the mods? I mean, these are mods, there can be literally anything in mods, caliber conversion kits included.

**Comment by smolbird** [comment]
> By the way, why do we do this in the mods? I mean, these are mods, there can be literally anything in mods, caliber conversion kits included.

Well Realistic Weapons Pack has very painfully obvious reasons for this change. Icecoon's Weapons Pack could potentially get away with retool kits of this sort, but note that a 6.54mm autofire kit was also obsoleted from it already. While it has fictional guns its still geared toward at least seeming consistent relative to mainline flavor and mechanics. 

Saying "it's a mod" is no excuse for not making a mod consistent with what its apparent objective is.

**Comment by Night-Pryanik** [comment]
> but note that a 6.54mm autofire kit was also obsoleted from it already.

Autofire kit was obsoleted from IWP incorrectly, for the same reason caliber conversion kits are being obsoleted from it now. It shouldn't be done in the first place. 

>Saying "it's a mod" is no excuse for not making a mod consistent with what its apparent objective is.

I continue to insist that mods are free to do anything that's not causing errors.

**Comment by smolbird** [comment]
Mods are free to be any flavor and level of consistency-with-mainline they wish to be, but they should at least be *internally* consistent.

**Comment by Night-Pryanik** [comment]
> but they should at least be internally consistent.

Yes, of course. But what's inconsistent in having caliber conversion kit in a mod, even without autofire kit?

**Comment by kevingranade** [comment]
I could see moving the *generic* conversion kits to the *generic* gun mod, but all the gun *addition* mods only differ from the rest of the game in what guns are available, so by default if we change the way something works in the main game, we should change it in the gun addition mods too.
